### DISCUSSION TITLE
Why does "podman generate systemd" generate "Type=forking" (when run without --new)?

### DISCUSSION THREAD

**Discussion by eriksjolund** [containers/podman#15053] [General]
### "systemd generate systemd" (without --new)

`podman generate systemd` without __--new__ generates services with `Type=forking`

#### test 1
```
$ podman run --name test1 -d docker.io/library/nginx
d36d96292cffde340a42cc7e0f115874dfe351b47d873cbc24cc39bb63363449
$ podman generate systemd --name test1 | grep Type=
Type=forking
$
```

#### test 2

```
$ podman run --name test2 --sdnotify=conmon  -d docker.io/library/nginx
04ec43f0d8688eb60beb29bb0f3c18a673d329ad71d0eda7ebef278de0e05766
$ podman generate systemd --name test2 | grep Type=
Type=forking
```

#### test 3

```
$ podman run --name test3 --sdnotify=container  -d docker.io/library/nginx
dc4fa293ec6d0ff32bcdaaa7abafb042d3f8471485c4cdbfc6ddfdf8be58c374
$ podman generate systemd --name test3 | grep Type=
Type=forking
```

### "systemd generate systemd" (with --new)

`podman generate systemd` with __--new__ generates services with `Type=notify`

```
$ podman generate systemd --name test1 --new | grep Type=
Type=notify
$ podman generate systemd --name test2 --new | grep Type=
Type=notify
$ podman generate systemd --name test3 --new | grep Type=
Type=notify
```

### Discussion

Could it be that  `Type=forking` was chosen because of the default value for the 
 for the `podman run` option [__--sdnotify__](https://docs.podman.io/en/latest/markdown/podman-run.1.html#sdnotify-container-conmon-ignore)? The  default value is currently  _container_.

That would explain the design choice for __test 1__ but not for __test 2__ and __test 3__.

I'm starting to think that the default value for __--sdnotify__ should be changed from _container_ to _conmon_.

It would then be possible for  `systemd generate systemd` to generate `Type=notify` for __test 1__, __test 2__, __test 3__.

**Comment by eriksjolund** [comment]
I found a quote from the PR 
https://github.com/containers/podman/pull/10557
that added `Type=notify` support for `--new`:

"_but it doesn't really work well since we're competing with the default
sdnotify mode set during container creation._"

**Reply by vrothberg** [reply]
Yes, that is it.  With `podman start` we cannot control the sdnotify-mode of the container.

**Comment by mheon** [comment]
@vrothberg PTAL
